### PR TITLE
Api refresh

### PR DESCRIPTION
**Pull request by RabebOthmani**
In this PR: I updated Number Insight to use the new APIs and implement new functionalities(Advanced, CNAM). 
There was some naming violations too that I'm fixing along the way.

### CODE REVIEW

File: Nexmo.Api/NumberInsight.cs [hunk 1]
```diff
@@ -1,139 +1,194 @@
...
-            public string ported { get; set; }
-            public string reachable { get; set; }
-            public string roaming { get; set; }
-            public string roaming_country_code { get; set; }
-            public string roaming_network_code { get; set; }
-        }
-
+            /// </summary>
+            public CarrierInfo original_carrier { get; set; }
+            public string CallerName { get; set; }
```
Comment (judy2k) [inline Nexmo.Api/NumberInsight.cs]: What's the reason for the difference in naming convention here? eg: `original_carrier` vs `CallerName`

In the Java client lib we map the names from the json convention (snake_case) to the Java convention (camelCase). Not sure if you want to do this in the dotnet lib.

Comment (RabebOthmani) [reply Nexmo.Api/NumberInsight.cs]: Good point! I missed some of the old props, I'll update everything to follow the C# convention.

File: Nexmo.Api.Test.Unit/NumberInsightTest.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ namespace Nexmo.Api.Test.Unit
     internal class NumberInsightTest : MockedWebTest
     {
         [Test]
-        public void should_send_basic_ni_request()
+        public void ShouldSendBasicNIRequest()
```
Comment (smithrobs) [inline Nexmo.Api.Test.Unit/NumberInsightTest.cs]: If we're going to switch to upcase from underscore in test naming, we might as well be consistent and change them across the test suite. Personally I prefer underscore as it feels closer to spaces and more human-readable for test names in particular.

File: Nexmo.Api/NumberInsight.cs [hunk 3]
```diff
@@ -1,139 +1,194 @@
...
+using Nexmo.Api.Request;
+using System.Threading.Tasks;
+
+namespace Nexmo.Api
+{
+    public static class NumberInsight
+    {
+        public enum CallerType
+        {
+            Unkown,
```
Comment (smithrobs) [inline Nexmo.Api/NumberInsight.cs]: Should be "Unknown"

File: Nexmo.Api/NumberInsight.cs [hunk 4]
```diff
@@ -142,39 +197,53 @@ public static NumberInsightBasicResponse RequestBasic(NumberInsightBasicRequest
...
+        }
+
+        public static NumberInsightAdvancedResponse RequestAdvanced( NumberInsightAdvancedRequest request, Credentials creds = null)
+        {
+            var response = ApiRequest.DoPostRequest(ApiRequest.GetBaseUriFor(typeof(NumberVerify), "/ni/advanced/json"), request, creds);
+
+            return JsonConvert.DeserializeObject<NumberInsightAdvancedResponse>(response.JsonResponse);
+        }
+
+        //public static NumberInsightAdvancedResponse RequestAdvancedAsync(NumberInsightAdvancedRequest request, Credentials creds = null)
```
Comment (smithrobs) [inline Nexmo.Api/NumberInsight.cs]: Is this a TODO? If so, please comment in code as such and add issue so we can circle back and resolve later.

Comment (RabebOthmani) [reply Nexmo.Api/NumberInsight.cs]: There's already an issue for this, it's commented until testing properly is done.

File: Nexmo.Api.Test.Unit/NumberInsightTest.cs (hunk 2 above)
Comment (smithrobs) [reply Nexmo.Api.Test.Unit/NumberInsightTest.cs]: So we're moving to upcase, then? 😄

**Comment by RabebOthmani** [comment]
@smithrobs Totally agree that these changes should be based off/merged into V3 Branch

**Comment by smithrobs** [comment]
Closing due to updated PR that goes against `3.0`

**Pull request closed without merging**